app/components/WaitTimer/WaitTimer.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+
+function WaitTimer(props) {
+
+  const appointmentTime = props.startTime;
+  var textLabel: string = '';
+
+  const calculateTime = () => {
+
+    const currentTime = new Date().getTime();
+    var estimatedTime: number;
+
+    if (appointmentTime > currentTime) {
+      // future schedule appointment case
+      estimatedTime = appointmentTime - currentTime;
+      textLabel = 'Starting: ';
+    } else {
+      // past scheduled appointment case
+      estimatedTime = currentTime + (currentTime - (currentTime + appointmentTime));
+      textLabel = 'Waiting: ';
+    }
+
+    let timeLeft = {};
+
+    timeLeft = {
+      dd: Math.floor(estimatedTime / (1000 * 60 * 60 * 24)),
+      hh: Math.floor((estimatedTime / (1000 * 60 * 60)) % 24),
+      mm: Math.floor((estimatedTime / 1000 / 60) % 60),
+      ss: Math.floor((estimatedTime / 1000) % 60),
+    };
+    return timeLeft;
+  };
+
+  const [timeLeft, setTimeLeft] = useState(calculateTime());
+
+  useEffect(() => {
+    setTimeout(() => {
+      setTimeLeft(calculateTime());
+    }, 1000);
```

Comment:
need to clear timeout if element is destroyed. Please search the entire project and find a solution.